torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -764,16 +764,15 @@ std::shared_ptr<at::cuda::CUDAEvent> ProcessGroupNCCL::CUDAEventCache::create(
   at::cuda::CUDAEvent* event = nullptr;
   {
     std::lock_guard<std::mutex> lock(cacheMutex_);
-    auto events = eventsArray_[timing ? 1 : 0];
+    auto& events = eventsArray_[timing ? 1 : 0];
     if (!events.empty()) {
-      event = events.back();
-      events.pop_back();
+      event = events.front();
+      events.pop_front();
```

Comment:
any reason to prefer one order over the other?  (why did you switch from stack to fifo?)

Reply:
because fifo will return the oldest event first which makes it safter?